### PR TITLE
updated the comment and the return for error code 501 in views.py

### PR DESCRIPTION
**Pull request by HadiAlshakhori**

### CODE REVIEW

File: pontoon/machinery/views.py [hunk 1]
```diff
@@ -62,8 +63,8 @@ def translation_memory(request):
                 suggestions[entry['target']].update(entry)
             suggestions[entry['target']]['count'] += 1
     except DataError as e:
-        # Catches 'argument exceeds the maximum length of 255 bytes' Error
-        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+        #  Catches the error number in the argument that have the status number 501
```
Comment (mathjazz) [inline pontoon/machinery/views.py]: Please leave the original comment, it's still valid.

File: pontoon/machinery/views.py [hunk 2]
```diff
@@ -62,8 +63,8 @@ def translation_memory(request):
...
+        return HttpResponse(status=501, reason='Not Implemented:{error}'.format(error=e))
```
Comment (mathjazz) [inline pontoon/machinery/views.py]: Add space between `Not Implemented:` and `{error}`.

**Comment by HadiAlshakhori** [comment]
Thank  Matjaz for the comments , I have pushed the updates from your comment .

Thank you

File: pontoon/machinery/views.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 from pontoon.base import utils
 from pontoon.base.models import Locale, TranslationMemoryEntry
+from django.http import HttpResponse
```
Comment (mathjazz) [inline pontoon/machinery/views.py]: We already import from `django.http` a few lines up, so the line 10 should read like:
`from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse`

File: pontoon/machinery/views.py [hunk 4]
```diff
@@ -63,7 +64,7 @@ def translation_memory(request):
             suggestions[entry['target']]['count'] += 1
     except DataError as e:
         # Catches 'argument exceeds the maximum length of 255 bytes' Error
-        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+        return HttpResponse(status=501, reason='Not Implemented: {error}'.format(error=e))
```
Comment (mathjazz) [inline pontoon/machinery/views.py]: Whoops, my bad - text should not be in the `reason` attribute, it should be passed as the first positional argument, followed by `status`.

`return HttpResponse('Not Implemented: {error}'.format(error=e), status=501)`

File: pontoon/machinery/views.py [hunk 5]
```diff
@@ -62,8 +62,9 @@ def translation_memory(request):
                 suggestions[entry['target']].update(entry)
             suggestions[entry['target']]['count'] += 1
     except DataError as e:
-        # Catches 'argument exceeds the maximum length of 255 bytes' Error
-        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+
```
Comment (mathjazz) [inline pontoon/machinery/views.py]: Nit: Please remove the blank line.

File: pontoon/machinery/views.py [hunk 6]
```diff
@@ -62,8 +62,8 @@ def translation_memory(request):
                 suggestions[entry['target']].update(entry)
             suggestions[entry['target']]['count'] += 1
     except DataError as e:
-        # Catches 'argument exceeds the maximum length of 255 bytes' Error
-        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+        # Catches argument exceeds the maximum length of 255 bytes' Error
```
Comment (mathjazz) [inline pontoon/machinery/views.py]: Nit: `'` got lost somewhere.

**Pull request merged**